src/framework/parsers/glb-parser.js

```diff
@@ -2022,139 +2014,165 @@ const loadImageAsync = (gltfImage, index, bufferViews, urlBase, registry, option
+            } else if (gltfImage.hasOwnProperty('uri')) {
                 // uri specified
                 if (isDataURI(gltfImage.uri)) {
-                    loadTexture(gltfImage.uri, null, getDataURIMimeType(gltfImage.uri), null);
-                } else {
-                    loadTexture(ABSOLUTE_URL.test(gltfImage.uri) ? gltfImage.uri : path.join(urlBase, gltfImage.uri), null, null, { crossOrigin: 'anonymous' });
+                    return loadTexture(gltfImage, gltfImage.uri, null, getDataURIMimeType(gltfImage.uri), null);
                 }
+                return loadTexture(gltfImage, ABSOLUTE_URL.test(gltfImage.uri) ? gltfImage.uri : path.join(urlBase, gltfImage.uri), null, null, { crossOrigin: 'anonymous' });
             } else if (gltfImage.hasOwnProperty('bufferView') && gltfImage.hasOwnProperty('mimeType')) {
                 // bufferview
-                loadTexture(null, bufferViews[gltfImage.bufferView], gltfImage.mimeType, null);
-            } else {
-                // fail
-                callback('Invalid image found in gltf (neither uri or bufferView found). index=' + index);
+                return loadTexture(gltfImage, null, bufferViews[gltfImage.bufferView], gltfImage.mimeType, null);
             }
+
+            // fail
+            return Promise.reject(new Error(`Invalid image found in gltf (neither uri or bufferView found). index=${i}`));
+        });
+
+        if (postprocess) {
+            promise = promise.then((textureAsset) => {
+                postprocess(gltfImage, textureAsset);
+                return textureAsset;
+            });
         }
-    });
+
+        result.push(promise);
+    }
+
+    return result;
 };
 
-// load textures using the asset system
-const loadTexturesAsync = (gltf, bufferViews, urlBase, registry, options, callback) => {
-    if (!gltf.hasOwnProperty('images') || gltf.images.length === 0 ||
-        !gltf.hasOwnProperty('textures') || gltf.textures.length === 0) {
-        callback(null, []);
-        return;
+// create gltf textures. returns an array of promises that resolve to texture assets.
+const createTextures = (gltf, images, options) => {
+    const result = [];
+
+    if (!gltf?.images?.length || !gltf?.textures?.length) {
+        return result;
     }
 
     const preprocess = options?.texture?.preprocess;
-    const processAsync = options?.texture?.processAsync || ((gltfTexture, gltfImages, callback) => {
-        callback(null, null);
-    });
+    const processAsync = options?.texture?.processAsync;
     const postprocess = options?.texture?.postprocess;
 
-    const assets = [];        // one per image
-    const textures = [];      // list per image
+    const seenImages = new Set();
 
-    let remaining = gltf.textures.length;
-    const onLoad = (textureIndex, imageIndex) => {
-        if (!textures[imageIndex]) {
-            textures[imageIndex] = [];
+    for (let i = 0; i < gltf.textures.length; ++i) {
+        const gltfTexture = gltf.textures[i];
```

Comment:
Don't you prefer:

```
for (const texture of gltf.texture) {
```

Reply:
I've made them `array.map`s instead, thanks!